### ISSUE TITLE
Selector with 'default' case disables check for missing default in outer case statement

### ISSUE THREAD

**Issue by dsnt02518** [rodjek/puppet-lint#829]
The following code passes puppet-lint 2.3.6 without errors:
```
# Class description
class test {
  case $::operatingsystem {
    'centos': {
      $variable = $::operatingsystemmajrelease ? {
        '7' => 'value1',
        default => 'value2',
      }
      # Note there's no default for the case statement here.
    }
  }
}
```

It looks like the check for a 'default' case is catching the selector default case in the wrong context.
